Configured hibernate

sjchmiela opened the pull request:
Finally accomplished this. The changes:
- maven imports newest hibernate package, along with MySQL JDBC driver, Java Transaction API package and Hibernate-C3P0 – connection pool manager,
- hibernate is configured with `hibernate.cfg.xml` file from `/resources` and mapping files,
- hibernate connection properties can be overridden with environment variables `DB_USERNAME`, `DB_PASSWORD`, `DB_PORT` and `DB_HOST`,
- `HibernateUtils` class with static methods manages storing session factory,
- for the time being the `HomeController` tries to fetch Students explicitly with Hibernate query.

aalices committed:
Create models' skeletons

sjchmiela committed:
Added Hibernate dependency. Fetches from Maven repository.

sjchmiela committed:
Added Hibernate dependency. Fetches from Maven repository.

sjchmiela committed:
Created empty Hibernate configuration and HibernateUtils class.

sjchmiela committed:
Created empty Hibernate configuration and HibernateUtils class.

sjchmiela committed:
Added mysql-connector-java dependency for Hibernate connection to MySQL database.

sjchmiela committed:
Added mysql-connector-java dependency for Hibernate connection to MySQL database.

sjchmiela committed:
Provided default configuration for connection properties and overriding based on environment variables.

sjchmiela committed:
Provided default configuration for connection properties and overriding based on environment variables.

sjchmiela committed:
Added mappings for every concrete model class.

sjchmiela committed:
Added mappings for every concrete model class.

sjchmiela committed:
Fixed old initializer.

sjchmiela committed:
Fixed old initializer.

sjchmiela committed:
Moved hibernate configuration to a place which maven picks up.

sjchmiela committed:
Moved hibernate configuration to a place which maven picks up.

sjchmiela committed:
Added JTA, required by Hibernate.

sjchmiela committed:
Added JTA, required by Hibernate.

sjchmiela committed:
Fixed property names of hibernate configuration overriding.

sjchmiela committed:
Fixed property names of hibernate configuration overriding.

sjchmiela committed:
Tried to list students from database. Somehow the student isn't mapped.

sjchmiela committed:
Tried to list students from database. Somehow the student isn't mapped.

sjchmiela committed:
Added C3P0 pool manager.

sjchmiela committed:
Added C3P0 pool manager.

sjchmiela committed:
Move HBM settings to the place from which they get copied to final module.

sjchmiela committed:
Move HBM settings to the place from which they get copied to final module.

sjchmiela committed:
It works with FQN.

sjchmiela committed:
It works with FQN.

sjchmiela committed:
With older versions it works without FQN.

sjchmiela committed:
With older versions it works without FQN.

aalices committed:
Setup Spring Boot - resources not working

sjchmiela committed:
Use the newest libraries versions

sjchmiela committed:
Use the newest libraries versions

sjchmiela committed:
Use default values for properties in hibernate.cfg.xml

sjchmiela committed:
Use default values for properties in hibernate.cfg.xml

sjchmiela committed:
Configure Hibernate properly based on configuration file

Previous method of building ServiceRegistry set only properties from hibernate.cfg.xml, it didn't include mappings. The new method includes and parses mappings.

sjchmiela committed:
Configure Hibernate properly based on configuration file

Previous method of building ServiceRegistry set only properties from hibernate.cfg.xml, it didn't include mappings. The new method includes and parses mappings.

aalices committed:
Change to Spring Boot

aalices commented on colon-app/src/main/java/pl/edu/agh/to/app/controller/HomeController.java:
```diff
@@ -1,17 +1,27 @@
 package pl.edu.agh.to.app.controller;
```
I would be grateful if you look at new changes (Spring Boot) and make this file consistent :)

sjchmiela replied:
Or maybe you could merge your branch with master, so I can merge master with my branch, and then make pull request? I don't know which option is more gitty. :)

aalices replied:
sure!

sjchmiela committed:
Merge branch 'configure_hibernate' of https://github.com/margg/colon into configure_hibernate

# Conflicts:
#	colon-app/src/main/java/pl/edu/agh/to/app/controller/HomeController.java
#	colon-app/src/main/java/pl/edu/agh/to/app/model/Group.java
#	colon-app/src/main/java/pl/edu/agh/to/app/model/Task.java

sjchmiela committed:
Add missing class attributes and reorder properties

Added missing class attributes needed for other subsystems like eg. file_path in Solution. Reordered some of the properties of classes and edited their not-nullity – eg. Administrator has to have username and password, but doesn't have to have first or last name, so the required fields should be before the additional ones.

sjchmiela committed:
Configure Hibernate not to drop the tables but to update them if possible

sjchmiela committed:
Use proper hibernate mapping namespace

sjchmiela committed:
Swap swapped not-nullity of columns
